apps/007_ofsted_market_analysis/app.py

```diff
@@ -2,113 +2,498 @@
 import pandas as pd
 import plotly.express as px
 import matplotlib
+from pyodide.http import open_url
+
+def needs_coder(row, field):
+    if row[field] == 'Y':
+        return 'Yes'
+    elif (int(row[field])) > 0:
+        return 'Yes'
+    else:
+        return 'No'
+
+def plot_chart(data_frame, var_x, var_y, var_color, var_title, var_barmode, var_cdm=None, var_cat_orders=None):
+    fig = px.bar(data_frame,
+        x = var_x,
+        y = var_y,
+        color = var_color,
+        title = var_title,
+        barmode=var_barmode,
+        color_discrete_map=var_cdm,
+        category_orders=var_cat_orders)
+    st.plotly_chart(fig)
+
+
+data1 = open_url('https://raw.githubusercontent.com/Paggers92/PATCh_test/main/ofsted_benchmarking_app/Local%20authorities%20and%20regions.csv')
+regions = pd.read_csv(data1)
 
 st.markdown("[![Foo](https://github.com/data-to-insight/patch/blob/main/docs/img/contribute.png?raw=true)](https://www.datatoinsight.org/patch) \
              [![Foo](https://github.com/data-to-insight/patch/blob/main/docs/img/viewthecodeimage.png?raw=true)](https://github.com/data-to-insight/patch/blob/main/apps/007_ofsted_market_analysis/app.py)")
 
 # Title and description
-st.title('Ofsted Market Analysis - small providers by region')
-st.write('This tool analyses a list of social care providers as provided by Ofsted.')
-st.write('The user can filter on one or multiple regions, as well as a maximum threshold of settings per owner, to view a list of socal care settings that fit these criteria.')
+st.title('Ofsted Market Analysis')
+st.write('This tool analyses a list of social care settings providers as provided by Ofsted.')
+st.write('Use the sidebar selectors to filter by geography and provider type.')
+st.write('Click on the tabs below to view different breakdowns for the selected filters.')
 
 # Takes "Ofsted Social care providers list for LAs"
-uploaded_files = st.file_uploader('Upload Ofsted social care providers list:', accept_multiple_files=False)
+uploaded_files = st.file_uploader('Upload Ofsted social care providers list:', accept_multiple_files=True)
+
 
 if uploaded_files:
-    if uploaded_files.name[-4:] == '.csv':
-        loaded_files = {uploaded_files.name: pd.read_csv(uploaded_files)}
-    else:
-        loaded_files = {uploaded_files.name: pd.read_excel(uploaded_files)}
-    
-    for file in loaded_files.items():
-        st.write('Displaying "' + file[0] + '"') # Print the key from the dictionary for this file
-        #st.dataframe(file[1]) # Print the dataframe from the dictionary for this file
 
-    # Access the dataframe for the file
-    df = list(loaded_files.values())[0]
+    loaded_files = {uploaded_file.name: pd.read_csv(uploaded_file) for uploaded_file in uploaded_files}
+
+    files_dict = {}
+
+    for name,file in loaded_files.items():

```

Comment:
I'm trying to think of a way to do this without a loop but I'm coming up blank, something to think about for a later version though